dali/operators/image/color/color_twist.h

```diff
@@ -20,148 +20,91 @@
   }
-
- private:
-  float brightness_;
 };
 
 class Contrast : public ColorAugment {
  public:
-  void operator() (float * matrix) override {
-    for (int i = 0; i < nDim - 1; ++i) {
-      for (int j = 0; j < nDim - 1; ++j) {
-        matrix[i * nDim + j] *= contrast_;
-      }
-      matrix[i * nDim + nDim - 1] = matrix[i * nDim + nDim - 1] * contrast_ +
-                                    (1 - contrast_) * 128.f;
-    }
-  }
-
   void Prepare(Index i, const OpSpec &spec, const ArgumentWorkspace * ws) override {
-    contrast_ = spec.GetArgument<float>("contrast", ws, i);
+    auto contrast = spec.GetArgument<float>("contrast", ws, i);
+    matrix_ = mat_t(contrast);
+    matrix_(nDim-1, nDim-1) = 1.;
+    for (int r = 0; r < nDim-1; ++r) matrix_(r, nDim-1) = (1 - contrast) * 128.f;
   }
-
- private:
-  float contrast_;
 };
 
 class Hue : public ColorAugment {
  public:
-  void operator() (float * matrix) override {
-    float temp[nDim*nDim];  // NOLINT(*)
-    for (int i = 0; i < nDim * nDim; ++i) {
-        temp[i] = matrix[i];
-    }
-    const float U = cos(hue_ * M_PI / 180.0);
-    const float V = sin(hue_ * M_PI / 180.0);
-
+  void Prepare(Index i, const OpSpec &spec, const ArgumentWorkspace * ws) override {
     // Single matrix transform for both hue and saturation change. Matrix taken
     // from https://beesbuzz.biz/code/hsv_color_transforms.php. Derived by
     // transforming first to HSV, then do the modification, and transfom back to RGB.
-
-    const float const_mat[] = {.299, .587, .114, 0.0,
-                               .299, .587, .114, 0.0,
-                               .299, .587, .114, 0.0,
-                               .0,   .0,   .0,   1.0};
-
-    const float U_mat[] = { .701, -.587, -.114, 0.0,
-                           -.299,  .413, -.114, 0.0,
-                           -.300, -.588, .886,  0.0,
-                            .0,    .0,   .0,    0.0};
-
-    const float V_mat[] = { .168,   .330, -.497, 0.0,
-                           -.328,   .035,  .292, 0.0,
-                           1.25, -1.05,  -.203, 0.0,
-                            .0,    .0,    .0,   0.0};
-    // The last row stays the same so we update only nDim - 1 rows
-    for (int i = 0 ; i < nDim - 1; ++i) {
-      for (int j = 0; j < nDim; ++j) {
-        float sum = 0;
-        for (int k = 0; k < nDim; ++k) {
-          sum += temp[k * nDim + j] * (const_mat[i * nDim + k] +
-                                       U_mat[i * nDim + k] * U +
-                                       V_mat[i * nDim + k] * V);
-        }
-        matrix[i * nDim + j] = sum;
-      }
-    }
-  }
-
-  void Prepare(Index i, const OpSpec &spec, const ArgumentWorkspace * ws) override {
-    hue_ = spec.GetArgument<float>("hue", ws, i);
+    auto hue = spec.GetArgument<float>("hue", ws, i);
+    const float U = cos(hue * M_PI / 180.0);
+    const float V = sin(hue * M_PI / 180.0);
+    const mat_t const_mat = mat_t({{.299, .587, .114, 0.0},
```

Comment:
How about moving this const_mat  and U_mat  to ColorAugment  as Saturation  uses it as well?

Reply:
I've moved all the matrices to ColorAugment, it seemed more consistent.